homeassistant/components/amcrest/helpers.py

```diff
@@ -1,13 +1,12 @@
 """Helpers for amcrest component."""
+from homeassistant.util import slugify
+
 from .const import DOMAIN
 
 
-def service_signal(service, ident=None):
-    """Encode service and identifier into signal."""
-    signal = f"{DOMAIN}_{service}"
-    if ident:
-        signal += f"_{ident.replace('.', '_')}"
-    return signal
+def service_signal(service, *args):
+    """Encode signal."""
+    return slugify("_".join([DOMAIN, service, *args]))
```

Comment:
I don't think the user is ever going to see this, does this need to be slugifyed ?

Reply:
It was an easy way to create a reasonably unique signal name. Is there a problem doing it this way?

Reply:
Does it still work if you take the `slugify` out?


return "_".join([DOMAIN, service, *args])

Reply:
Maybe you're missing what this function does. This integration can send a lot of different signals to the various platform modules. This function creates a unqiue signal name based on its parameters. It is only used internal to this integration. I don't see why this is a problem? Have you looked at how it is used within the integration?

Reply:
`slugify` is really slow. Frenck and I were able to validate a million entity IDs in 0.6s instead of 12s by no longer using `slugify`: https://github.com/home-assistant/core/pull/32137

Reply:
Thanks, and I can understand wanting to optimize. But in this use case this only happens when a camera goes offline or comes back online, or when a motion event is detected. In normal usage I can't see this happening more than once every minute per camera (for motion events. For offline/online it should almost never happen.) So even with six cameras that means not more than once every ten seconds. I measured it with `timeit`, including the regex `findall` call and `slugify`, and it came out to about 20 µsec. So let's say on a really slow machine it's 100 µsec. If 100 µsec / 10 sec is too much for a given installation then I think they have bigger problems to tackle.

As far as why I'm doing this, I was attempting to make the signal names be somewhat uniform. One of the items that goes into it is the camera's name (not its object ID.) But I guess there aren't any restrictions on what a signal name can be, just that they're unique. I already guarantee uniqueness of camera names, so I suppose I don't really need `slugify`. I'll remove it.